seshdash/utils/alert.py

```diff
@@ -16,92 +16,206 @@
-                if user.seshuser.phone_number and user.seshuser.on_call:
-                    sms_recipients.append(user.seshuser.phone_number)
-
-
-            logging.debug("emailing %s" %email_recipients)
-            #print "emailing %s"%recipients
-            # recipients = ["seshdash@gmail.com",]
-            
-            """ Creating an alert object """
-            alert_obj = Sesh_Alert.objects.create(
-                    site = site,
-                    alert=rule,
-                    date=timezone.now(),
-                    isSilence=False,
-                    emailSent=False,
-                    slackSent=False,
-                    smsSent=False,
-                    point_model=type(data_point).__name__ )
-            alert_obj.save()
-            
-            
-            # Set data point to point to alert
-            data_point.target_alert = alert_obj
-            data_point.save()
-            alerts = Sesh_Alert.objects.all()
+        # Get datapoint and value
+        data_point, real_value = get_alert_check_value(site, rule)
 
+        if data_point is not None and real_value is not None:
 
-            if rule.send_mail:
-                mail_sent = alertEmail(data_point,content,email_recipients)
-                alert_obj.emailSent = mail_sent
+            if check_alert(rule, real_value):
+                content = get_alert_content(site, rule, data_point, real_value)
+                mails, sms_numbers = get_recipients_for_site(site)
+          
+                alert_obj = alert_factory(site, rule, data_point)
+
+                if rule.send_mail:
+                     alert_obj.emailSent = alertEmail(data_point,content,mails)
             
-            if rule.send_sms:
-                sms_response = alertSms(data_point,content,sms_recipients)
-                alert_obj.smsSent = sms_response
+                if rule.send_sms:
+                     alert_obj.smsSent = alertSms(data_point,content,sms_numbers)
             
-            alert_obj.save()
+                alert_obj.save()
+
+
+
 
 def alertEmail(data_point,content,recipients):
     return send_mail("Alert email from seshdash",recipients,content)
 
+
+
 def alertSms(data_point,content,recipients):
     return send_sms(recipients, content)
 
+
+
+def get_alert_check_value(site, rule):
+    """ Returns the value to check for alert from latest data point """
+      
+    # if rule is valid
+    if '#' in rule.check_field:
+        model, field_name = rule.check_field.split('#')
+        latest_data_point = get_latest_data_point_for_rule(site, rule)
+        
+        if latest_data_point is not None:
+            data_point_value = getattr(latest_data_point, field_name)
+        else:
+            data_point_value = None
+            logging.error("No data points for %s", model)
+        
+        return latest_data_point, data_point_value
+
+    else:
+        logging.error('Invalid alert rule')
+        return False
```

Comment:
Better to throw an exception here as this code will already throw an exception that's misleading. it won't be able to unpack this into to values. See line:29